crates/starknet-server/src/api/json_rpc/mod.rs

```diff
@@ -0,0 +1,461 @@
+                    data: None,
+                },
+                err @ ApiError::ContractNotFound => RpcError {
+                    code: server::rpc_core::error::ErrorCode::ServerError(20),
+                    message: err.to_string().into(),
+                    data: None,
+                },
+                err @ ApiError::TransactionNotFound => RpcError {
+                    code: server::rpc_core::error::ErrorCode::ServerError(25),
+                    message: err.to_string().into(),
+                    data: None,
+                },
+                err @ ApiError::InvalidTransactionIndexInBlock => RpcError {
+                    code: server::rpc_core::error::ErrorCode::ServerError(27),
+                    message: err.to_string().into(),
+                    data: None,
+                },
+                err @ ApiError::ClassHashNotFound => RpcError {
+                    code: server::rpc_core::error::ErrorCode::ServerError(28),
+                    message: err.to_string().into(),
+                    data: None,
+                },
+                err @ ApiError::ContractError => RpcError {
+                    code: server::rpc_core::error::ErrorCode::ServerError(40),
+                    message: err.to_string().into(),
+                    data: None,
+                },
+                err @ ApiError::NoBlocks => RpcError {
+                    code: server::rpc_core::error::ErrorCode::ServerError(32),
+                    message: err.to_string().into(),
+                    data: None,
+                },
+                err @ ApiError::RequestPageSizeTooBig => RpcError {
+                    code: server::rpc_core::error::ErrorCode::ServerError(31),
+                    message: err.to_string().into(),
+                    data: None,
+                },
+                err @ ApiError::InvalidContinuationToken => RpcError {
+                    code: server::rpc_core::error::ErrorCode::ServerError(33),
+                    message: err.to_string().into(),
+                    data: None,
+                },
+                err @ ApiError::TooManyKeysInFilter => RpcError {
+                    code: server::rpc_core::error::ErrorCode::ServerError(34),
+                    message: err.to_string().into(),
+                    data: None,
+                },
+            }
+            .into(),
+        }
+    }
+}
+
+#[async_trait::async_trait]
+impl RpcHandler for JsonRpcHandler {
+    type Request = StarknetRequest;
+
+    async fn on_request(&self, request: Self::Request) -> ResponseResult {
+        info!(target: "rpc", "received method in on_request");
+        self.execute(request).await
+    }
+}
+
+/// This object will be used as a shared state between HTTP calls.
+/// Is simillar to the HttpApiHandler but is with extended functionality and is used for JSON-RPC methods
+#[derive(Clone)]
+pub struct JsonRpcHandler {
+    pub api: Api,
+}
+
+impl JsonRpcHandler {
```

Comment:
Instead of having a huge case matcher, can't it be separated into multiple traits?

Reply:
The case matcher can't be separated

Reply:
Certainly, but thinking about it more, this huge case matching might not be a problem as in other languages, since the compiler should check if all cases are covered. Resolving.